### PR TITLE
[Testing] Ban/DB overhaul

### LINKED ISSUE #8584: Identity Bans are broken
Apply them all you want but they don't force randomname/body 

Happening on live Sybil

### PR DESCRIPTION
**Pull request by tkdrg**
Let's declutter these awful tables from legacy junk and old snowflake. Basically what @MrStonedOne said in #8584 aka everything is now a job ban (renamed to "role ban")
- [x] Turn appearance bans into job bans (incl. migration)
- [x] Remove temp vs perma bans, make expiration_time allow NULL, and if expiration_time is NULL, perma, else, temp.
- [x] Remove admin ban, make that a field. 'Applies to admins', now we can have admin job bans.
- [x] Remove type, if role is null, its a server ban, else role/job ban.
- [x] Remove duration, it's not used, expiration_time is.
- [x] Remove the rounds column, as it is unused.
- [x] Ultimately remove bantype as it will be implied by the db entry.
- [x] ~~Rename job ban to role ban~~ Deferred due to laziness.
- [x] Write up something to alter the old tables to the new schema.
- [x] Write up some migration proc to convert old server/admin bans to the new schema
- [x] Fix the default schema to get rid of duration/bantype & add the admin field.
- [ ] Test everything (haha as if).

Fixes #8584 and probably a bunch of other issues that I can't be assed to find.
More importantly, it improves and reorganizes ban code, kills snowflake, deprecates the million old ban systems that we have in favor of db bans (though I might keep legacy server permabans around for people who just want a simple private server without the hassle of SQL, and I know mso has some plans for something along the lines of SQlite-based flat file DBs which would render that point moot), reduces the total amount of columns in the DB, and makes the dm code leaner and cleaner.

### CODE REVIEW

**Comment by optimumtact** [comment]
:+1: I remember when we last did this and all the bans disappeared.

**Comment by lzimann** [comment]
wew lads

**Comment by MrStonedOne** [comment]
The migration process would just be a few update queries that I could help write. 

The duration, is used for displaying the minutes the ban is for, but that is all, ban time calculation uses the expires field. So those rare cases could use http://dev.mysql.com/doc/refman/5.5/en/date-and-time-functions.html#function_timestampdiff to calculate it from when the ban was placed

**Comment by Bawhoppen** [comment]
lets hope for full ban removal :+1:

**Comment by Jordie0608** [comment]
RIP tkdrg's sanity

**Comment by tkdrg** [comment]
Jokes on you I was never sane all along
Anyway, the migration stuff is committed, now I just need to actually do the dm changes which will hopefully happen before the weekend

**Comment by tkdrg** [comment]
Okay, I'm done with the code for this. Still no testing whatsoever, though, but opinions/reviews are welcome.

File: SQL/ban_db_migration_2016-02-22.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+ALTER TABLE SS13_BAN CHANGE expiration_time expiration_time datetime DEFAULT NULL;
+ALTER TABLE SS13_BAN ADD applies_to_admins tinyint(1) DEFAULT NULL AFTER a_ip;
+
+UPDATE SS13_BAN SET job='appearance',expiration_time=NULL WHERE bantype = 'APPEARANCE_PERMABAN';
+UPDATE SS13_BAN SET job='appearance' WHERE bantype = 'APPEARANCE_TEMPBAN';
+
+UPDATE SS13_BAN SET job=NULL WHERE job='';
```
Comment (MrStonedOne) [inline SQL/ban_db_migration_2016-02-22.sql]: You might want to change the default on the job table from '' to null

**Comment by MrStonedOne** [comment]
I'll work on testing this out today and on monday.

**Comment by optimumtact** [comment]
Excitement intensifies

**Comment by optimumtact** [comment]
Hello
Hello
Come in

**Comment by Shadowlight213** [comment]
Rip pr

**Comment by duncathan** [comment]
db code literally killed him

**Comment by Bawhoppen** [comment]
still better than goofchem

yow im too good

**Comment by KorPhaeron** [comment]
@MrStonedOne 

Do you plan on salvaging this?

**Comment by optimumtact** [comment]
The code has no merge conflicts, so it's just blocked on the testing stage.

**Comment by MrStonedOne** [comment]
at some point, yes.

File: SQL/ban_db_migration_2016-02-22.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+UPDATE SS13_BAN SET expiration_time=NULL WHERE bantype = 'PERMABAN' OR bantype = 'JOB_PERMABAN' OR bantype = 'ADMIN_PERMABAN';
+
+UPDATE SS13_BAN SET applies_to_admins=1 WHERE bantype = 'ADMIN_TEMPBAN' OR bantype = 'ADMIN_PERMABAN';
+
+ALTER TABLE SS13_BAN DROP COLUMN bantype;
+ALTER TABLE SS13_BAN DROP COLUMN duration;
+ALTER TABLE SS13_BAN DROP COLUMN rounds;
```
Comment (MrStonedOne) [inline SQL/ban_db_migration_2016-02-22.sql]: Table names are case sensitive depending on configuration.

This should be `SS13_ban` not `SS13_BAN`

**Comment by MrStonedOne** [comment]
Getting issues where trying to ban is silently failing

**Comment by duncathan** [comment]
no big deal :^)

**Comment by KorPhaeron** [comment]
Guess this is dead then unless tkdrg comes back or mso makes his own PR

**Pull request closed without merging**